### ISSUE TITLE
"npm test" fails with EPERM at fs.exists() on Windows

### ISSUE THREAD

**Issue by seaoak** [hexojs/hexo#4290]
## Check List

Please check followings before submitting a new issue.

- [x] I have already read [Docs page](https://hexo.io/docs/) & [Troubleshooting page](https://hexo.io/docs/troubleshooting)
- [x] I have already searched existing issues and they are not help  to me
- [x] I examined error or warning messages and it's difficult to solve
- [ ] Using [the latest](https://github.com/hexojs/hexo/releases) version of Hexo (run `hexo version` to check)
- [x] Node.js is higher than 8.6.0

## Expected behavior

`npm test` command passes.

## Actual behavior

`npm test` command fails with `Error: EPERM: operation not permitted` as follows:

![SS20200503a hexo_npm_test_fail](https://user-images.githubusercontent.com/1160250/80908573-d8610a00-8d5b-11ea-8bc3-aa8a185c3fc9.png)

This error is caused by calling `fs.exists()` for `db.json`.
The file `db.json` is removed (unlinked) immediately before.

## How to reproduce?

```bash
$ git checkout master
$ npm test
```

## Environment

OS: Windows 10 64bit (rev.1909)
Node.js: v12.16.2, v14.1.0, v10.20.1
AntiVirus: none (use "Windows Defender")

Source codes:

```bash
$ git log --oneline --decorate -1
2b63f9fc (HEAD -> master, upstream/master) perf(post): simplify codeblock escape (#4254)
$
$ (cd ../hexo-util && git log --oneline --decorate -1)
f5d510a (HEAD -> master, upstream/master, origin/master, origin/HEAD) Merge pull request #199 from curbengh/cache-class
$
$ (cd ../hexo-fs && git log --oneline --decorate -1) | cat
c0a4b6c (HEAD -> master, upstream/master, origin/master, origin/HEAD) ci(appveyor): stop updating npm (#68)
$
```

**Comment by seaoak** [comment]
As a trial, I inserted delay (100ms) between `fs.unlink()` and `fs.exists()`.
The result is OK (two test cases of "Save database" pass).
So I think this issue might be related to race condition of file operation.

```diff
diff --git a/test/scripts/filters/save_database.js b/test/scripts/filters/save_database.js
index d6a38368..9180f75a 100644
--- a/test/scripts/filters/save_database.js
+++ b/test/scripts/filters/save_database.js
@@ -3,6 +3,12 @@
 const { exists, unlink } = require('hexo-fs');
 const Promise = require('bluebird');

+const sleep = async (period_ms) => {
+  return new Promise(resolve => {
+    setTimeout(() => resolve(), period_ms);
+  });
+};
+
 describe('Save database', () => {
   const Hexo = require('../../../lib/hexo');
   const hexo = new Hexo();
@@ -17,7 +23,8 @@ describe('Save database', () => {
     const exist = await exists(dbPath);
     exist.should.be.true;

-    unlink(dbPath);
+    await unlink(dbPath);
+    await sleep(100);
   });

   it('do nothing if hexo is not initialized', async () => {
```

**Comment by seaoak** [comment]
Before `hexo-fs` `3.0.0`, the `EPERM` error at `fs.access()` in `fs.exists()` was discarded.
I think this is because this issue comes to the surface now.

**Comment by seaoak** [comment]
This issue seems to be resolved by the PR #4322

**Comment by SukkaW** [comment]
> This issue seems to be resolved by the PR #4322

Ref: https://github.com/hexojs/warehouse/pull/81
